router/http/src/main/scala/io/buoyant/router/http/MethodAndHostIdentifier.scala

```diff
@@ -24,21 +26,28 @@ case class MethodAndHostIdentifier(
   private[this] def mkPath(path: Path): Dst.Path =
     Dst.Path(prefix ++ path, baseDtab(), Dtab.local)
 
-  def apply(req: http.Request): Future[(Dst, http.Request)] = req.version match {
+  def apply(req: http.Request): Future[RequestIdentification[Request]] = req.version match {
     case http.Version.Http10 =>
       Future.value(
-        (mkPath(Path.Utf8("1.0", req.method.toString) ++ suffix(req)), req)
+        new IdentifiedRequest[Request](mkPath(Path.Utf8("1.0", req.method.toString) ++ suffix(req)), req)
```

Comment:
i'd guess the type annotation isn't needed here.

ALso I'd be inclined to have a `val path = mkPath(...)` so that you can fit this all one line with a bit more clarity